### PR TITLE
[63] Implement Error boundary

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- implement error handler hooks with unit test
- I will write about how to use it in notion later

## Demo

- https://dev.modela.miello.dev/login login with false password
- https://dev.modela.miello.dev/signup/actor signup with already used email
- others error

### CODE REVIEW

File: apps/web/src/common/hooks/useErrorHandler/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const handleError = (
+    error: any,
+    overrideMessage?: { [key: string]: string },
+  ) => {
+    if (error instanceof AxiosError) {
+      if (error.response) {
+        const { data, status } = error.response
+
+        console.log(status)
+        console.log(status === 401)
```
Comment (miello) [inline apps/web/src/common/hooks/useErrorHandler/index.ts:26]: Consider remove `console.log`

File: apps/web/src/common/hooks/useErrorHandler/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          showError?.(
+            overrideMessage?.[409] || 'เกิดข้อขัดแย้งกับข้อมูลที่มีอยู่',
+          )
+        } else if (status === 500) {
+          if (overrideMessage?.[500]) showError?.(overrideMessage?.[500])
+          else replace('/500')
+        } else {
+          replace('/500')
+        }
+        console.log(`${status}: ${JSON.stringify(data)}`)
```
Comment (miello) [inline apps/web/src/common/hooks/useErrorHandler/index.ts:48]: Consider remove `console.log`

**Review by miello — CHANGES_REQUESTED** [review]
There are unit tests error which came from `useUserData`, please mock next/router there also

File: apps/web/jest.setup.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+jest.doMock('common/hooks/useErrorHandler', () => ({
+  useErrorHandler: () => ({ handleError: jest.fn() }),
+}))
+
+jest.doMock('next/router', () => ({
+  useRouter: () => ({
+    push: jest.fn(),
+    replace: jest.fn(),
+  }),
+}))
```
Comment (miello) [inline apps/web/jest.setup.js:10]: (Discussion)
Should we move file that relate to transforming data or global mock to another folder ? maybe `__mock__`

File: apps/web/src/modules/error/pages/FallbackPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        alignItems: 'center',
+        textAlign: 'center',
+        padding: '20px',
+        gap: '10px',
+      }}
```
Comment (miello) [inline apps/web/src/modules/error/pages/FallbackPage/index.tsx:18]: move to `styled.ts` ?

File: apps/web/src/modules/error/pages/FallbackPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    >
+      <Typography variant={isMobile ? 'h6' : 'h4'}>
+        ขออภัย เราไม่พบหน้าที่คุณต้องการค้นหา
+        <SentimentVeryDissatisfiedOutlined
+          style={{ transform: 'translateY(2px)', marginLeft: '5px' }}
```
Comment (miello) [inline apps/web/src/modules/error/pages/FallbackPage/index.tsx:23]:
```suggestion
          sx={{ transform: 'translateY(2px)', marginLeft: '5px' }}
```

File: apps/web/jest.setup.js (hunk 3 above)
Comment (PKhing) [reply apps/web/jest.setup.js:10]: like `__mock__/jest.setup.js` ?

Comment (miello) [reply apps/web/jest.setup.js:10]: Yes

File: apps/web/src/common/hooks/useErrorHandler/index.ts (hunk 2 above)
Comment (PKhing) [reply apps/web/src/common/hooks/useErrorHandler/index.ts:48]: I intend to write this `console.log` (for logging errors).

Comment (miello) [reply apps/web/src/common/hooks/useErrorHandler/index.ts:48]: Oh okay, resolved

**Review by miello — APPROVED** [review]

**Pull request merged**